### PR TITLE
Additional content

### PR DESCRIPTION
**Pull request by lynxx131**
This is an attempt to port Rundas01's PR #129 to the current dev-1.18 branch.

I have a few problems / questions though...

### 1. I was unsure of how to port the distillation stuff as I see material registrations like this:

```
public static Material WoodTar = AntimatterAPI.register(Material.class, new Material(Ref.ID, "wood_tar", 0x28170b, NONE))
            .asFluid(0,0,true,new DistillationProduct[]{new DistillationProduct(Creosote,"fluid",720),new DistillationProduct(Phenol,"fluid",180),
                    new DistillationProduct(Benzene,"fluid",360),new DistillationProduct(Toluene,"fluid",180)},1440)
```

I'm assuming that the `DistillationProduct` arguments being passed to asFluid are to define distillation outputs? If so I am unsure of what the correct way to do this would be against the newer branch.

### 2. Had issues figuring out how to port the distillery class over:

```
public class DistilleryLoader {
    public static void init() {
        AntimatterAPI.all(Material.class).forEach(m -> {
            if (!(m.canDistill)) return;
            for(int i=0;i<m.distillsInto.length;i++){
                DISTILLING.RB().ii(INT_CIRCUITS.get(i).setNoConsume()).fi(m.getLiquid(fluidamount(m.amount))).fo(m.distillsInto[i].getLiquid(m.amount[i])).add(m.getDensity(),20);
            }
        });
    }
```

I'm assuming that MaterialTags would be the way to go in the absence of the canDistill property?

### 3. Couldn't find replacement `Material.getHardness()` when porting ExtruderLoader class

In lines like:

```
EXTRUDING.RB().ii(of(INGOT.getMaterialTag(r),1),of(GregTechData.ShapeRing,1).setNoConsume()).io(RING.get(r,4)).add(r.getHardness(),30);
```

I did:

```
long duration = Math.max(r.getMass(), 1);
EXTRUDING.RB().ii(of(INGOT.getMaterialTag(r),1),of(GregTechData.ShapeRing,1).setNoConsume()).io(RING.get(r,4)).add(duration,30);
```

not sure if this is correct or not.

Not had time to troubleshoot this yet but

I'm keen to see this mod grow as it's my primary blocker to moving over to 1.18 and I do have a development background (just not a huge amount of experience with MC modding) so would like to chip in where possible.

### CODE REVIEW

**Comment by lynxx131** [comment]
My last issue is in Miscellaneous.java:

I can't figure out how to port the plantball recipes:

```
provider.addItemRecipe(output, Ref.ID, "plantball", "misc", "has_wrench", provider.hasSafeItem(WRENCH.getTag()), PlantBall, of(
                'C', Tags.Items.CROPS
        ), "CCC", "C C", "CCC");
```

The `Tags` class is `net.minecraftforge.common.Tags` but I presume due to the desire to maintain compatibility with both Forge and Fabric I can't just directly reference something in `net.minecraftforge` so would appreciate some guidance here.

**Comment by Vliro** [comment]
If it's common, you can't access these no. You can probably just create a tag, since any tag with the same resource location point to the same thing so that's fine.

**Comment by Trinsdar** [comment]
Yah use the TagUtils.getForgelikeTag method

**Comment by lynxx131** [comment]
Sorted the tag stuff and those recipes work now just need to figure out my issues in the opening comment.

Still understanding how the machine processing side of things works.

**Comment by Trinsdar** [comment]
Well one thing is I haven't actually merged the pr in antimatter yet that this depends on,  as i was waiting for rundas to reply to one of my comments, but he seems to have disappeared

**Comment by Trinsdar** [comment]
As this depends on this: https://github.com/GregTech-Intergalactical/AntimatterAPI/pull/188

File: common/src/main/java/muramasa/gregtech/loader/machines/ChemicalReactorLoader.java [hunk 1]
```diff
@@ -6,58 +6,180 @@
...
+    }
+
+    private static void alkanoles_2_alkenes(){
+        alkanole_2_alkene(Ethanol,Ethylene,1,1);
+        alkanole_2_alkene(Propanol,Propene,1,1);
+        alkanole_2_alkene(Butane,Butene,1,1);
+        alkanole_2_alkene(Butanediol,Butadiene,1,2);
+    }
+
+    private static void alkane_2_alkanole(Material alkane, Material alkanole, int mol_alk, int mol_ox, RecipeIngredient... circuit){
```
Comment (Trinsdar) [inline common/src/main/java/muramasa/gregtech/loader/machines/ChemicalReactorLoader.java]: Rename this and any other methods that don't use camel case to camel case please

**Review by Trinsdar — CHANGES_REQUESTED** [review]

File: common/src/main/java/muramasa/gregtech/loader/machines/ChemicalReactorLoader.java (hunk 1 above)
Comment (lynxx131) [reply common/src/main/java/muramasa/gregtech/loader/machines/ChemicalReactorLoader.java]: This should be sorted now

**Review by Trinsdar — APPROVED** [review]

**Comment by Trinsdar** [comment]
is that everything?

**Comment by lynxx131** [comment]
No I need to sort the distillery stuff still, currently working on that.

Also I think I've found the use for the `distillationAmount` argument that you questioned in the Antimatter API PR.

I think that argument is meant to be used as the fluid input amount for the distillation recipes, just figuring out how to get that value through to the distillery class.

I could probably just set all the distillery recipes to use 1 bucket of fluid input for the sake of getting a working implementation.

**Comment by lynxx131** [comment]
Was thinking maybe this https://github.com/GregTech-Intergalactical/AntimatterAPI/pull/194 for passing the distillation input amount into the distillery.

**Comment by Trinsdar** [comment]
@lynxx131 ok gregtech has been updated with your antimatter pr

**Comment by lynxx131** [comment]
I think the PR is done now.

I feel like the distillation implementation could be further improved but for now this PR provides a working implementation.

**Comment by Vliro** [comment]
Great work!

**Pull request merged**